src/de/error.rs

```diff
@@ -31,11 +31,14 @@ pub enum Error {
     /// The end of the BSON input was reached too soon.
     EndOfStream,
 
-    /// An invalid timestamp was encountered while decoding.
-    InvalidTimestamp(i64),
+    /// An invalid datetime was encountered while decoding.
+    InvalidDateTime {
+        /// The key at which an unexpected/unsupported datetime was encountered.
+        key: String,
 
-    /// An ambiguous timestamp was encountered while decoding.
-    AmbiguousTimestamp(i64),
```

Comment:
This case was never actually reachable as per [the docs for `chrono`](https://docs.rs/chrono/0.4.19/chrono/offset/trait.TimeZone.html#method.timestamp_opt).